src/mlpack/methods/ann/image_functions/bilinear_function.hpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file bilinear_function.hpp
+ * @author Kris Singh
+ *
+ * Definition and implementation of the bilinear interpolation function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_IMAGE_FUNCTIONS_INTERPOLATION_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_IMAGE_FUNCTIONS_INTERPOLATION_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The bilinear interpolation function
+ *
+ */
+class BiLinearFunction
+{
+ public:
+  BiLinearFunction(
+      const size_t inRowSize,
+      const size_t inColSize,
+      const size_t scale):
```

Comment:
I think it isn't difficult to support real (`double`) scales. Moreover, it is a good idea to support changing of the aspect ratio.
So, I suggest 4 input parameters: `inRowSize`, `inColSize`, `outRowSize`, `outColSize`.